src/Tutorial.ts

```diff
@@ -0,0 +1,438 @@
+                    tooltipPopUp = tooltipPopUp + "</div>"
+                    tooltipPopUp = tooltipPopUp + "<div class='tutorialInfoText'>"
+                        tooltipPopUp = tooltipPopUp + "<span>"+step.getText()+"</span>"
+                    tooltipPopUp = tooltipPopUp + "</div>"
+                    tooltipPopUp = tooltipPopUp + "<div class='tutorialInfoButtons'>"
+                        if(activeTutCurrentStep>1){
+                            tooltipPopUp = tooltipPopUp + "<button class='tutPreviousBtn' onclick='eagle.tutorial().tutButtonPrev()'>Previous</button>"
+                        }
+                        if(activeTutCurrentStep<activeTutStepsNo && step.getType() != TutorialStep.Type.Press){
+                            tooltipPopUp = tooltipPopUp + "<button class='tutNextBtn' onclick='eagle.tutorial().tutButtonNext()'>Next</button>"
+                        }
+                        tooltipPopUp = tooltipPopUp + "<span class='tutProgress'>"+ activeTutCurrentStep +" of "+activeTutStepsNo+"</span>"
+                        tooltipPopUp = tooltipPopUp + "<button class='tutEndBtn' onclick='eagle.tutorial().tutButtonEnd()'>Exit</button>"
+                    tooltipPopUp = tooltipPopUp + "</div>"
+                tooltipPopUp = tooltipPopUp + "</div>"
+
+            tooltipPopUp = tooltipPopUp + "</div>"
+        tooltipPopUp = tooltipPopUp + "</div>"
+
+        $('body').append(tooltipPopUp)
+    }
+
+    closeInfoPopUp = () : void =>{
+        $("#tutorialInfoPopUp").remove()
+        $(".tutorialHighlight").remove()
+    }
+
+    tutButtonNext = () : void => {
+        if(cooldown === false){
+            if(activeTutCurrentStep<activeTutStepsNo){
+                this.closeInfoPopUp()
+                activeTutCurrentStep ++
+                this.initiateTutStep('next')
+                this.startCooldown()
+            }else{
+                this.tutButtonEnd()
+            }
+        }
+    }
+
+    tutButtonPrev = () : void => {
+        if(cooldown === false){
+            if(activeTutCurrentStep>1){
+                this.closeInfoPopUp()
+                activeTutCurrentStep --
+                this.initiateTutStep('prev')
+                this.startCooldown()
+            }
+        }
+    }
+
+    tutButtonEnd = () : void => {
+        this.closeInfoPopUp()
+        $('body').off('keydown.tutEventListener');
+        $('.tutButtonListener').off('click.tutButtonListener').removeClass('tutButtonListener')
+    } 
+
+    tutPressStepListener = () : void => {
+        $('.tutButtonListener').off('click.tutButtonListener').removeClass('tutButtonListener')
+        this.tutButtonNext()
+    }
+}
+
+export class TutorialStep {
+    private title : string;
+    private text : string;
+    private type : TutorialStep.Type;
+    private waitType : Wait.Type;
+    private selector : () => void;
+    private preFunction : (eagle:Eagle) => void;
```

Comment:
Some comments of the purpose of/requirement for the forward and back funcs would be good.